analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -94,6 +94,27 @@ public override ProgramState[] PostProcess(SymbolicContext context)
                     };
                 }
             }
+            else if (context.Operation.Instance.AsObjectCreation() is { } objectCreation
```

Comment:
When I added the constraint, the lock-released tests started to fail and while I was tracking this down, I realized, that the logic for this rule was not right. I don't know what we want to do with this change as it is unrelated to the sprint.

This change actually does the same as what was done in line 84 above, just for a different overload.

Reply:
We can do the change, especially when you have the context. But I'd split it in another PR as it has nothing to do with Enum.
Here, just accept the problem with ToDo